### ISSUE TITLE
Delay the process of deleting a group

### ISSUE THREAD

**Issue by nilsreichardt** [SharezoneApp/sharezone-app#509]
Sometimes we get requests that someone accidentally deleted a course and ask if we could restore the data. Unfortunately, currently we can't do this because we immediately delete the data. The loss of a group can have disastrous consequences, like losing years of homeworks, files, information sheets and events.

What we could do (besides improve the UX like #151) is to delay the process of deletion.

This is what it would look like:
1. User deletes a group
2. We remove all users from this group
3. Disable joining the group (changing the group settings)
4. Schedule a delete request (maybe with Google Cloud Task)
5. Delete the course completely

In case of a recover, a user could send us a support ticket. We check (has the user the problem to recover the course) the request and if user is authorized we add enable joining the group and send the user a Sharecode. At this point, the user might need to some steps for themselves:
* Add the course again to a school class
* Old course members need to join again for themselves

A better solution (more advanced solution) would be to store the group list at the time of the deletion (just keeping the `Members` collection). When restoring the group, we automatically add all members from then to the group (edge case: a member could have been deleted meanwhile).

**Comment by Jonas-Sander** [comment]
Couldn't we add sth like a recycle bin to which it is moved and deleted after like 7 or 14 days?

**Comment by nilsreichardt** [comment]
Yes, this would also be solution 👍 This solution would also be scalable to all other features 👌
